server/src/termdb.filter.js

```diff
@@ -47,7 +47,8 @@ export async function getFilterCTEs(filter, ds, CTEname = 'f') {
 			// .CTEname: str
 		} else if (!item.tvs) {
 			throw `filter item should have a 'tvs' or 'lst' property`
-		} else if (!ds.cohort.termdb.q.termjsonByOneid(item.tvs.term.id)) {
+		} else if (item.tvs.term.id && !ds.cohort.termdb.q.termjsonByOneid(item.tvs.term.id)) {
```

Comment:
should be
```js
else if (item.tvs.term.id) {
   // some tvs such as geneVariant and samplelst don't have term id
  if !ds.cohort.termdb.q.termjsonByOneid(item.tvs.term.id)) throw 'invalid term id in tvs'			
}
```

Reply:
need to put "!ds.cohort.termdb.q.termjsonByOneid(item.tvs.term.id)" in the condition so that the "else if" below it won't be skipped